refactor/message_display

BarAgam opened the pull request.

barmako committed:
Refactored MasterMessagesViewModel.
Made it work against a single interactor that handles the app presentation logic as-well.
 As a result, a new interactor was implemented DisplayMessagesInteractor that took the place of the old Get and Sync messages interactor,.

 Another impact this change have was to take the select and read state out of the domain app and keeping it in it's own repositories.

 Refactored ContainerMessagesViewModel to work with a new version an interactor that's also manges presentation logic named DisplayUnreadMessageCountInteractor

barmako committed:
WIP
Display Selected Message usecase (interactor) implementation.
Still missing geometry data display.

barmako committed:
refactor/develop merge fixes.
SendUserLocationInteractor doesn't add UserLocation messages to MessagesRepository anymore.

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/112" target="_blank"><img src="https://app.review.ninja/52010471/pull/112/badge" alt="ReviewNinja"/>

barmako committed:
Rebase fixing.

barmako committed:
Bug fix - display interactor would update the same adapter twice with the same data.

barmako committed:
Changed both image and geo messages to hold their GeoEntity instead of it's ID.
Changed transformers accordingly.
Changed Details fragments to use the new information.
Changed detail fragments so that the message information is taken form their MessageContainerFragment using a target fragment.

barmako committed:
Optimized master fragment message display time

barmako committed:
transformer no longer stores geo entities by itself.
